### PR TITLE
Add ESP PumpDescription support

### PR DESCRIPTION
**Pull request by arthursoprana**

### CODE REVIEW

**Comment by prusse-martin** [comment]
You probably want to update [this filled case](https://github.com/ESSS/alfasim-sdk/blob/v0.13.0/tests/common_testing/alfasim_sdk_common_testing/filled_case_descriptions.py#L479-L488) with non default values to check if the conversion alfacase <-> case_description is working properly.

**Comment by prusse-martin** [comment]
I did just check and looks like the `EquipmentDescription` loader is implemented using an auto loader so it should work in production, the problem if updating that filled case is with `load_pump_equipment_description` that exist solely to appease the unit test gods.
That test do not check any production code... I am not sure but I think just removing that explicit loader will cause the test to fallback to auto loaders.
If/when you tackle this feel free to call me. I hate all this zombie code lingering around.

File: src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py [hunk 1]
```diff
@@ -1709,20 +1709,10 @@ def _generate_description(
...
-        "thermal_efficiency": get_scalar_loader(from_unit="-"),
-        "table": load_table_pump_description,
-        "speed_curve": load_speed_curve_description,
-        "speed_curve_interpolation_type": get_enum_loader(
-            enum_class=constants.InterpolationType
-        ),
-        "flow_direction": get_enum_loader(enum_class=constants.FlowDirection),
-    }
-
+    alfacase_to_case_description = get_case_description_attribute_loader_dict(
```
[the remark this answers is not in the document]
Comment (prusse-martin) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:1712]: My point is that this function is not used in production.
The new loader strategy do not need any explicit code (but we are not fully ported).

**Review by prusse-martin — APPROVED** [review]

**Comment by prusse-martin** [comment]
The merge conflict is just the "cog gard hash" =)

File: src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arthursoprana) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py]: Ok, I'll remove it, thanks

**Review by ggrbill — APPROVED** [review]

**Pull request merged**